### PR TITLE
feat: support com_qeury update sql request

### PR DESCRIPTION
**Pull request by dk-lockdown**
ref: https://github.com/cectc/dbpack/issues/104

### Ⅰ. Describe what this PR did

support com_query update sql request, such as `UPDATE /*+ XID('gs/svc/2612354882102353922') */ salaries SET salary = 20000 WHERE emp_no = 100003`, and `SqlUndoLog` should distinguish between binary protocol and text protocol. 

### Ⅲ. Why don't you add test cases (unit test/integration test)? 
there is integration test. refer to `test/sdb/distributed_transaction_test.go`, func `TestDistributedTransactionQueryRequest`

### Ⅳ. Describe how to verify it

see the ci reesult
image

### CODE REVIEW

File: pkg/dt/mysql_undo_executor.go [hunk 1]
```diff
@@ -60,11 +60,11 @@ func DeleteBuildUndoSql(undoLog *undolog.SqlUndoLog) string {
 	var sbCols, sbVals strings.Builder
```
Comment (bohehe) [inline pkg/dt/mysql_undo_executor.go:60]: Better to use `BuildDeleteUndoSql` as func name

File: pkg/dt/mysql_undo_executor.go [hunk 2]
```diff
@@ -224,25 +224,64 @@ func (executor MysqlUndoExecutor) queryCurrentRecords(tx proto.Tx) (*schema.Tabl
...
+		dataTable, _, err := tx.Query(context.Background(), selectSql)
+		if err != nil {
+			return nil, err
+		}
+		dt := dataTable.(*mysql.Result)
+		return schema.BuildTextRecords(tableMeta, dt), nil
+	}
+}
+
+func (executor MysqlUndoExecutor) buildCurrentRecordsPrepareSql(tableMeta schema.TableMeta, pkColumn string, pkValues []interface{}) string {
```
Comment (bohehe) [inline pkg/dt/mysql_undo_executor.go]:
```suggestion
func (executor MysqlUndoExecutor) buildCurrentRecordsForPreparedSql(tableMeta schema.TableMeta, pkColumn string, pkValues []interface{}) string {
```

File: pkg/dt/mysql_undo_executor.go [hunk 3]
```diff
@@ -224,25 +224,64 @@ func (executor MysqlUndoExecutor) queryCurrentRecords(tx proto.Tx) (*schema.Tabl
...
-
 	inCondition := misc.MysqlAppendInParam(len(pkValues))
-	selectSql := fmt.Sprintf(SelectSqlTemplate, b.String(), tableMeta.TableName, pkName, inCondition)
-	dataTable, _, err := tx.ExecuteSql(context.Background(), selectSql, pkValues...)
-	if err != nil {
-		return nil, err
+	return fmt.Sprintf(SelectSqlTemplate, b.String(), tableMeta.TableName, pkColumn, inCondition)
+}
+
+func (executor MysqlUndoExecutor) buildCurrentRecordsQuerySql(tableMeta schema.TableMeta, pkColumn string, pkValues []interface{}) string {
```
Comment (bohehe) [inline pkg/dt/mysql_undo_executor.go]:
```suggestion
func (executor MysqlUndoExecutor) buildCurrentRecordsForQuerySql(tableMeta schema.TableMeta, pkColumn string, pkValues []interface{}) string {
```

File: pkg/dt/undolog/undo_log.go [hunk 4]
```diff
@@ -22,6 +22,10 @@ import (
 )
 
 type SqlUndoLog struct {
+	// IsBinary binary protocol or text protocol, com_stmt_execute corresponds to
+	// binary protocol, com_query corresponds to text protocol.
```
Comment (bohehe) [inline pkg/dt/undolog/undo_log.go]:
```suggestion
	// binary protocol (prepared statement), com_query corresponds to text protocol (text satement).
```

**Review by bohehe — APPROVED** [review]
LGTM

**Review by bohehe — APPROVED** [review]

**Review by hzliangbin — APPROVED** [review]

**Comment by hzliangbin** [comment]
LGTM

**Review by fatelei — APPROVED** [review]
LGTM

**Pull request merged**